src/PolyhedralGeometry/iterators.jl

```diff
@@ -323,3 +325,43 @@ function homogenized_matrix(x::SubObjectIterator{<:RayVector}, v::Number = 0)
     end
     return matrix_for_polymake(x; homogenized=true)
 end
+
+function homogenized_matrix(x::AbstractVector{<:PointVector}, v::Number = 1)
+    if v != 1
+        throw(ArgumentError("PointVectors can only be (re-)homogenized with parameter 1, please convert to a matrix first."))
+    end
+    return stack((homogenize(x[i], v) for i in 1:length(x))...)
+end
+function homogenized_matrix(x::AbstractVector{<:RayVector}, v::Number = 0)
+    if v != 0
+        throw(ArgumentError("RayVectors can only be (re-)homogenized with parameter 0, please convert to a matrix first."))
+    end
+    return stack((homogenize(x[i], v) for i in 1:length(x))...)
+end
+
+homogenized_matrix(::SubObjectIterator, v::Number) = throw(ArgumentError("Content of SubObjectIterator not suitable for homogenized_matrix."))
+
+unhomogenized_matrix(x::SubObjectIterator{<:RayVector}) = matrix_for_polymake(x)
+
+unhomogenized_matrix(x::AbstractVector{<:PointVector}) = throw(ArgumentError("unhomogenized_matrix only meaningful for RayVectors"))
+
+_ambient_dim(x::SubObjectIterator) = Polymake.polytope.ambient_dim(x.Obj)
+
+################################################################################
+######## Unify matrices
+################################################################################
+
+# vector-like types: (un-)homogenized_matrix -> matrix_for_polymake
+# linear types: linear_matrix_for_polymake 
+# affine types: affine_matrix_for_polymake
+const AbstractCollection = Dict{UnionAll, Union}([(PointVector, AnyVecOrMat),
+                                                    (RayVector, AnyVecOrMat),
+                                                    (LinearHalfspace, Union{AbstractVector{<:Halfspace}, SubObjectIterator{<:Halfspace}, AnyVecOrMat}),
+                                                    (LinearHyperplane, Union{AbstractVector{<:Hyperplane}, SubObjectIterator{<:Hyperplane}, AnyVecOrMat}),
+                                                    (AffineHalfspace, Union{AbstractVector{<:Halfspace}, SubObjectIterator{<:Halfspace}, Tuple{<:AnyVecOrMat, <:Any}}),
+                                                    (AffineHyperplane, Union{AbstractVector{<:Hyperplane}, SubObjectIterator{<:Hyperplane}, Tuple{<:AnyVecOrMat, <:Any}})])
```

Comment:
I'd recommend avoid such a deep nesting, use the pairs syntax, and use trailing commas for future extensions
```suggestion
const AbstractCollection = Dict(
    PointVector => AnyVecOrMat,
    RayVector => AnyVecOrMat,
    LinearHalfspace => Union{AbstractVector{<:Halfspace}, SubObjectIterator{<:Halfspace}, AnyVecOrMat},
    LinearHyperplane => Union{AbstractVector{<:Hyperplane}, SubObjectIterator{<:Hyperplane}, AnyVecOrMat},
    AffineHalfspace => Union{AbstractVector{<:Halfspace}, SubObjectIterator{<:Halfspace}, Tuple{<:AnyVecOrMat, <:Any}},
    AffineHyperplane => Union{AbstractVector{<:Hyperplane}, SubObjectIterator{<:Hyperplane}, Tuple{<:AnyVecOrMat, <:Any}},
)
```